processor/src/decorators/mod.rs

```diff
@@ -278,6 +282,62 @@ where
 
         Ok(())
     }
+
+    /// Pushes the value and depth flags of a leaf indexed by `key` on a Sparse Merkle tree with
+    /// the provided `root`.
+    ///
+    /// The Sparse Merkle tree is tiered, meaning it will have leaf depths in `{16, 32, 48, 64}`.
+    /// The depth flags define the tier on which the leaf is located.
+    ///
+    /// The operand stack is expected to be arranged as follows:
+    /// - key, 4 elements.
+    /// - root of the Sparse Merkle tree, 4 elements.
+    ///
+    /// After a successful operation, the advice stack will look as follows:
+    /// - boolean flag set to `1` if the depth is `16` or `32`.
+    /// - boolean flag set to `1` if the depth is `16` or `48`.
+    /// - value word; will be zeroed if the tree don't contain a mapped value for the key.
+    ///
+    /// # Errors
+    /// Will return an error if:
+    /// - The provided Merkle root doesn't exist on the advice provider
+    /// - If, at depth `64`, the node value is not zero, and there is no associated leaf value
+    fn inject_smtget(&mut self) -> Result<(), ExecutionError> {
+        // fetch the arguments from the operand stack
+        let key = [self.stack.get(3), self.stack.get(2), self.stack.get(1), self.stack.get(0)];
+        let root = [self.stack.get(7), self.stack.get(6), self.stack.get(5), self.stack.get(4)];
+
+        // fetch the leaf depth from the Merkle store
+        // an empty tree will return depth `0`, but we always point to the minimum tier (16)
+        const TREE_DEPTH: Felt = Felt::new(64);
```

Comment:
This one was declared as a constant so it will be zero runtime overhead as it will be allocated in the static space of the program, and passed as reference to the advice provider.

`From::from` is not a `const fn`, so we can't use it there.